src/appleseed/renderer/modeling/scene/objectinstance.cpp

```diff
@@ -83,7 +83,7 @@ bool uses_alpha_mapping(const MaterialArray& materials)
     {
         if (materials[i])
         {
-            if (materials[i]->has_alpha_map())
+            if (materials[i]->has_alpha_map() && !materials[i]->has_uniform_opacity())
                 return true;
 
             if (const ShaderGroup* sg = materials[i]->get_uncached_osl_surface())
```

Comment:
Can we move this test just before if (materials[i]->has_alpha_map() ...) to avoid any issue with OSL materials?